x-pack/plugins/observability_solution/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -126,36 +143,58 @@ function getTransactionItem(
   transaction: WaterfallTransaction,
   linkedChildrenCount: number = 0
 ): IWaterfallTransaction {
+  const spanLinks: SpanLink[] = [];
+  for (let i = 0; i < (transaction[SPAN_LINKS_TRACE_ID]?.length ?? 0); i++) {
+    spanLinks.push(
+      dedot(
+        {
+          'trace.id': transaction[SPAN_LINKS_TRACE_ID]?.[i],
+          'span.id': transaction[SPAN_LINKS_SPAN_ID]?.[i],
```

Comment:
this isn't idea if these two arrays are not equal lengths. If they are not equal lengths, it means the map between the trace.ids and span.ids is wrong. I add a `?? "0"` to line 152 so it doesn't blow up, but is there a better way to handle this?